src/macros.php

```diff
@@ -189,6 +189,46 @@
     });
 }
 
+if (! Collection::hasMacro('sectionBy')) {
+    /*
+     * Splits a collection into sections grouped by a given key.
+     *
+     * @param string $sectionBy
+     *
+     * @return \Illuminate\Support\Collection
+     */
+    Collection::macro('sectionBy', function ($sectionByKey) {
```

Comment:
This logic feels somewhat heavy and unreadable to me.

Could we accomplish the same results by using a `groupBy` first and than a `map` to get the desired format?